pkg/engine/variables/operator/anyin.go

```diff
@@ -134,6 +134,14 @@ func anySetExistsInArray(key []string, value interface{}, log logr.Logger, anyNo
 		}
 		return false, isAnyIn(key, valueSlice)
 
+	case int, int32, int64, float32, float64, bool:
```

Comment:
Is the requirement only for numbers? What about strings?

Reply:
The case for Strings have already present [(link)](https://github.com/kyverno/kyverno/blob/751b978dd000134ccda0107c14d218df074f7927/pkg/engine/variables/operator/anyin.go#L145) . That's the bug, there was no case for when like where key is array of int and value is int. All other possible combinations are already present.